src/Orc.Automation/ScenarioManagement/Models/AutomationScenario.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace Orc.Automation.ScenarioManagement;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Catel;
+
+public class AutomationScenario
+{
+    private readonly List<AutomationScenarioStep> _startedSteps;
+    private readonly List<AutomationScenarioStep> _finishedSteps;
+    private readonly Stack<AutomationScenarioStep> _runningSteps;
+    private readonly List<IAutomationScenarioLogger> _loggers = new();
+
+    public AutomationScenario(string name, string description = "")
+    {
+        Argument.IsNotNullOrEmpty(() => name);
+
+        Name = name;
+        Description = description;
+
+        _startedSteps = new List<AutomationScenarioStep>();
+        _finishedSteps = new List<AutomationScenarioStep>();
+        _runningSteps = new Stack<AutomationScenarioStep>();
+    }
+
+    public string Name { get; }
+    public string Description { get; }
+    public string? Suite { get; set; }
+
+    public IReadOnlyList<AutomationScenarioStep> StartedSteps => _startedSteps;
```

Comment:
Should RunningSteps also be exposed?

Reply:
for now - i don't see appliance for it outside